tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py

```diff
@@ -216,6 +216,8 @@ def _fit(self,
         # TODO: calculate max emb concat layer size and use 1st layer as that value and 2nd in between number of classes and the value
         if params.get('layers', None) is not None:
             layers = params['layers']
+            if isinstance(layers, tuple):
+                layers = list(layers)
```

Comment:
Is this a bandaid to a bigger problem? Will every model need to keep this in mind and code logic to handle? Is there a way to generically fix this?

Reply:
Yes, if a model needs a list as input and is using hyperopt as the search algorithm, this is needed. It's hard to find a way to generically handle it, as we don't know if the input is intended to be a list but got converted to tuple by hyperopt, or it's intended to be a tuple originally.

Reply:
ah I see, that makes sense regarding the ambiguity.